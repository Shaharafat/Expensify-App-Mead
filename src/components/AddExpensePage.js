import React from "react";
import { connect } from "react-redux";
import { startAddExpenses } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";

export class AddExpensePage extends React.Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(expense) {
    this.props.startAddExpenses(expense);
    this.props.history.push("/");
  }
  
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpenses: (expenses) => dispatch(startAddExpenses(expenses)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
