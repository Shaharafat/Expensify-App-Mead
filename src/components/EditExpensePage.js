import React from "react";
import { connect } from "react-redux";
import { editExpense, removeExpense } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";

export class EditExpensePage extends React.Component {
  constructor(props) {
    super(props);
    this.editExpense = this.editExpense.bind(this);
    this.removeExpense = this.removeExpense.bind(this);
  }

  editExpense(expense) {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  }

  removeExpense() {
    this.props.removeExpense(this.props.expense);
    this.props.history.push("/");
  }

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.editExpense} />
        <button onClick={this.removeExpense}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      (expense) => expense.id === props.match.params.id
    ),
  };
};

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (expense) => dispatch(removeExpense(expense)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
