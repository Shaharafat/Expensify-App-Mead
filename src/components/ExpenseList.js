import React from 'react'
import {connect} from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'
const ExpenseList = (props) => (
  <div>
    Expense List
    {
      props.expenses.map(expense => (
        <ExpenseListItem key={expense.id} {...expense} />
      ))
    }
  </div>
)

const mapStateToProps = (state) => {
  return {
    // expenses: state.expenses,
    // filters: state.filters
    expenses: selectExpenses(state.expenses,state.filters),
  }
}

export default connect(mapStateToProps)(ExpenseList)
