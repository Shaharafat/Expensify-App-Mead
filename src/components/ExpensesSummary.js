import React from "react";
import { connect } from "react-redux";
import {Link} from 'react-router-dom'
import numeral from "numeral";
import getVisibleExpenses from "../selectors/expenses";
import getTotalValues from "../selectors/expense-total";

export const ExpensesSummary = ({ length, total }) => {
  const expenseWord = length > 1 ? "expenses" : "expense";
  const formattedTotal = numeral(total / 100).format("$0,0.00");
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          <span>{length} </span>
          {expenseWord} totaling of
          <span> {formattedTotal} </span>
        </h1>
        <div class="page-header__actions">
          <Link className="button" to="/create" >Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    length: visibleExpenses.length,
    total: getTotalValues(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
