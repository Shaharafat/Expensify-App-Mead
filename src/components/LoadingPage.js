import React from 'react'

const LoadingPage = () => {
  return ( 
    <div className="loader">
    <img className="loader-image" src="/images/loader.gif" alt="loading image"/>
    </div>
   );
}
 
export default LoadingPage;