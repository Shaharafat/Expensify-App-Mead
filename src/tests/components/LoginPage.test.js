import React from 'react'
import {shallow} from 'enzyme'
import { LoginPage } from '../../components/LoginPage'

// ðŸ§ª snapshot for login
test('should setup login page', () => {
  const wrapper = shallow(<LoginPage/>)
  expect(wrapper).toMatchSnapshot()
})

test('should should call startLogin on button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
})