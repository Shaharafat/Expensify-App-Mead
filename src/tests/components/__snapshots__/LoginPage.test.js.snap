// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should setup login page 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
